### PR TITLE
feat: add todos_api package

### PR DESCRIPTION
**Pull request by Limbou**
## Description

 Add:
- todos_api package
- todos_api interface 
- todo model class
- todo model tests
- todos_api github actions workflow

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by orestesgaolin** [comment]
I haven't had time to look too closely, but couple of things to consider:

- each package should have its own workflow, let me show it to you on a call
- it may be easier for a reviewer to split these 3 packages into 3 separate PRs chained into each other, this way you only see diff for a particular package

**Review by jorgecoca — APPROVED** [review]

File: packages/todos_api/lib/src/models/todo.dart [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+///
+/// Contains a [title], [description] and [id], in addition to a [isCompleted]
+/// flag.
+///
+/// If an [id] is provided, it cannot be empty. If no [id] is provided, one
+/// will be generated.
+///
+/// [Todo]s are immutable and can be copied using [copyWith], in addition to
+/// being serialized and deserialized using [toJson] and [fromJson]
+/// respectively.
```
Comment (orestesgaolin) [inline packages/todos_api/lib/src/models/todo.dart:20]: I like the documentation but this is a bit too verbose for a practical use. In real project we would rather describe the "domain" usage of the given model. The technical information about copyWith or fields that a given object posses are easy to discover. The use case for a class or model, however, can sometimes be not obvious just by looking at its name. Anyway, it's fine to keep it like that.

File: packages/todos_api/test/src/todo_test.dart [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        },
+        throwsA(isA<AssertionError>()),
+      );
+    });
+
+    test('two objects with the same fields are equal', () {
+      final todoItem = Todo(id: itemId, title: itemTitle);
+      final todoItem2 = Todo(id: itemId, title: itemTitle);
+      expect(todoItem, equals(todoItem2));
+    });
```
Comment (orestesgaolin) [inline packages/todos_api/test/src/todo_test.dart:36]: My preferred way to test for value equality would compare 3 objects: 2 that should be equal and 1 that's different. This way you check for positive and negative case.

```suggestion
    test('supports value equality', () {
      final todoItem = Todo(id: itemId, title: itemTitle);
      final todoItem2 = Todo(id: itemId, title: itemTitle);
      final todoItem3 = Todo(id: itemId, title: itemTitle, isCompleted: false);
      expect(todoItem, equals(todoItem2));
      expect(todoItem, isNot(equals(todoItem3)));
    });
```

File: packages/todos_api/test/src/todos_api_test.dart [hunk 3]
```diff
@@ -0,0 +1,6 @@
+// ignore_for_file: prefer_const_constructors
+import 'package:test/test.dart';
+
+void main() {
```
Comment (orestesgaolin) [inline packages/todos_api/test/src/todos_api_test.dart:4]: I suspect this file is linked to the abstract class `TodosApi`? Let's not keep empty test files

**Comment by orestesgaolin** [comment]
The title of the PR is no longer valid after the splitting

**Review by orestesgaolin — APPROVED** [review]

**Pull request merged**